logger/logger.go

```diff
@@ -0,0 +1,47 @@
+package logger
+
+import (
+	"fmt"
+	"sync"
+)
+
+var once sync.Once
+var log *Logger
+
+// Logger provided simple logger.
+type Logger struct {
+	verbose bool
+}
+
+// Setup configures logger
+func Setup(verbose bool) {
+	once.Do(func() {
+		log = &Logger{
+			verbose,
+		}
+	})
```

Comment:
Instead of using `sync.Once`, we should use `sync.Mutex` as otherwise, this function will initialize the global logger only the first time it is invoked, which is not the desired behavior

Reply:
Reverted

Reply:
but I see that you're still using the `if log == nil` which essentially does the same thing as `sync.Once`.

My point was to not make the logger a singleton, as it is now, but rather to just guard its initialization via a mutex.

The short version is, we don't need the if clause

Reply:
I thought we needed a singleton here. And manage it with method like SetVerose. Ok, will fix it.